### PR TITLE
1610 extract search order

### PR DESCRIPTION
**Pull request by fredkingham**

### CODE REVIEW

**Comment by fredkingham** [comment]
fixes #1610

File: search/subrecord_discoverable.py [hunk 1]
```diff
@@ -284,8 +284,22 @@ def get_model_fields(self):
             "which fields do you want to get off the model"
         )
 
+    def sort_fields(self, fields):
+        """
+        If the fields attribute is declared, use the order specified.
```
Comment (davidmiller) [inline search/subrecord_discoverable.py:289]: This method doesn't do what it's docstring says it does. Instead, we make sorting decisions in get fields.

File: search/search_rules.py [hunk 2]
```diff
@@ -116,6 +116,29 @@ def widget_descriptions(cls, user):
         )}
 
 
+class AbstractLocationMixin(object):
```
Comment (davidmiller) [inline search/search_rules.py:119]: What is the point of this class ?

File: search/subrecord_discoverable.py [hunk 3]
```diff
@@ -284,8 +284,22 @@ def get_model_fields(self):
...
+        return sorted(fields, key=lambda x: x.get_display_name())
+
     def get_fields(self):
-        """ Get all fields, if
+        """
+        Get all fields,
+        if they are declared as a `fields` attribute use that in the order
+        they are declared.
+
+        otherwise use the model fields, sorted by `sort_model_fields`
```
Comment (davidmiller) [inline search/subrecord_discoverable.py:301]: Not what the method is called

File: search/subrecord_discoverable.py [hunk 4]
```diff
@@ -354,13 +377,12 @@ def get_schemas(cls, user):
     def get_fields_for_schema(self):
```
Comment (davidmiller) [inline search/subrecord_discoverable.py:377]: What is the point of this method ?

**Pull request merged**